### PR TITLE
MFL API Fix

### PR DESCRIPTION
**Pull request by hannestschofenig**
This PR addresses https://github.com/hannestschofenig/mbedtls/issues/209 

The goal was also to better align the API (as noted here https://github.com/hannestschofenig/mbedtls/issues/130). However, the upstream Mbed TLS code has no harmonized API signature for the extension handling between the client and the server code either. Hence, the currently used API signature is consistent with the approach selected for upstream Mbed TLS in the sense that it is equally inconsistent. 

In a future attempt to reduce the code size the extension handling code should be moved out of the client and server files. This would offer the possibility to do some harmonization.

### CODE REVIEW

File: library/ssl_tls13_client.c [hunk 1]
```diff
@@ -1754,7 +1754,13 @@ static int ssl_client_hello_write_partial( mbedtls_ssl_context* ssl,
 #endif /* MBEDTLS_SSL_ALPN */
 
 #if defined(MBEDTLS_SSL_MAX_FRAGMENT_LENGTH)
-    ssl_write_max_fragment_length_ext( ssl, buf, end, &cur_ext_len );
+    if( ( ret = ssl_write_max_fragment_length_ext( ssl, buf,
+                                                   end - buf, &cur_ext_len )  ) != 0 )
```
Comment (hanno-becker) [inline library/ssl_tls13_client.c]:
```suggestion
    if( ( ret = ssl_write_max_fragment_length_ext( ssl, buf,
                                                   (size_t)( end - buf ), 
                                                   &cur_ext_len )  ) != 0 )
```
Some compilers could complain about the implicit signed to unsigned cast otherwise.

File: library/ssl_tls13_server.c [hunk 2]
```diff
@@ -3067,35 +3067,37 @@ static int ssl_client_hello_postprocess( mbedtls_ssl_context* ssl,
...
 {
     unsigned char *p = buf;
 
-    *olen = 0;
-    if( ( ssl->handshake->extensions_present & MAX_FRAGMENT_LENGTH_EXTENSION )
-        == 0 )
-    {
-        return( 0 );
-    }
-
```
Comment (hanno-becker) [inline library/ssl_tls13_server.c]: Why was this removed?

Comment (hannestschofenig) [reply library/ssl_tls13_server.c]: This was a mistake and I added it back

**Review by hanno-becker — APPROVED** [review]
